Add more descriptive personal course page

bigjr-mkkong opened the pull request:
This PR close issue #10 
Now the personal course page can display more information(include personal scheduler name, course name, and quarter in correct format)

Before:
![image](https://github.com/ucsb-cs156-w24/proj-courses-w24-5pm-3/assets/48128050/d88dac8d-ace3-431a-b4d0-4ffb22689f64)

After:
[http://proj-course-michael-dev.dokku-07.cs.ucsb.edu](http://proj-course-michael-dev.dokku-07.cs.ucsb.edu)
![image](https://github.com/ucsb-cs156-w24/proj-courses-w24-5pm-3/assets/48128050/6f5545fa-d436-4778-b592-00355827c81d)

bigjr-mkkong committed:
First commit for more descriptive course page

bigjr-mkkong committed:
Fixed a bug inside controller and wrote some tests to gain 100% coverage

bigjr-mkkong committed:
Fixed string format problem

bigjr-mkkong committed:
Clear mutation test and fix a format issue

bigjr-mkkong committed:
Minor fix for passing format checker

bigjr-mkkong committed:
Minor change to fix format check error

bigjr-mkkong committed:
Fix PScourse entity for format error

bigjr-mkkong committed:
Try to fix format error inside PSCourseController.java

bwr02 committed:
ran mvn git-code-format:format-code

bwr02 committed:
Merge branch 'main' into michael-coursesPage

chloeqta reviewed (dismissed):
LGTM

bwr02 committed:
Merge branch 'main' into michael-coursesPage

AnikaArora reviewed (dismissed):
Please add a small description in your PR. Also, add a "before" screenshot too so we can see exactly how your code changed the original version.

bwr02 committed:
Merge branch 'main' into michael-coursesPage

bigjr-mkkong commented:
> Please add a small description in your PR. Also, add a "before" screenshot too so we can see exactly how your code changed the original version.

I've updated my PR with a small description and before-after screenshots to clarify the changes I've made in this PR, please consider to merge.

bwr02 reviewed (dismissed):
LGTM

tallyhawley commented on src/main/java/edu/ucsb/cs156/courses/controllers/PSCourseController.java:
```diff
@@ -147,6 +148,23 @@ public ArrayList<PSCourse> postCourses(
       throw new IllegalArgumentException("class exists in schedule");
     }
 
+    String quarter_QYY;
```
Why are we storing the quarter in QYY format? There is a helper on the frontend to convert from YYYYQ to QYY, just keep it in YYYYQ on the backend for consistency.

bigjr-mkkong replied:
Fixed it, please consider to merge into main branch

bigjr-mkkong committed:
Changed backend quarter to be more consistent -- Not Ready to be Merged

There are some peoblems with frontend test code within this commit, it is not ready to be
merge into main branch

bigjr-mkkong committed:
Fixed frontend tests and re-format code

bwr02 reviewed (approved):
LGTM

bwr02 committed:
Merge branch 'main' into michael-coursesPage

sophiattran reviewed (approved):
lgtm

tallyhawley reviewed (approved):
Looks good to me:)
